client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,695 @@
+package zio.raft.client
+
+import zio._
+import zio.stream._
+import zio.raft.protocol._
+import scodec.bits.ByteVector
+import java.time.Instant
+
+/**
+ * Main RaftClient implementation using functional state-based approach.
+ * 
+ * Each connection state (Disconnected, Connecting, Connected) handles messages
+ * differently and transitions to other states based on the protocol.
+ * 
+ * Session management is integrated directly into the state machine.
+ */
+class RaftClient private (
+  stateRef: Ref[ClientState],
+  pendingOperationsRef: Ref[Map[Nonce, SessionOperation]],
+  serverRequestTrackerRef: Ref[ServerRequestTracker],
+  retryManager: RetryManager,
+  zmqTransport: ZmqClientTransport,
+  config: ClientConfig,
+  actionQueue: Queue[ClientAction]
+) {
+  
+  /**
+   * Internal method to initiate connection to the cluster.
+   * Called automatically on client startup.
+   */
+  private def initiateConnection(): Task[SessionId] = 
+    for {
+      // Validate capabilities
+      _ <- validateCapabilities(config.capabilities)
+      
+      // Transition to Connecting state
+      _ <- transitionTo(ClientState.Connecting(config.capabilities, nonce = None, addresses = config.clusterAddresses, currentAddressIndex = 0))
+      
+      // Generate nonce for session creation
+      nonce <- Nonce.generate()
+      
+      // Create promise for session creation
+      promise <- Promise.make[SessionError, SessionId]
```

Comment:
no need for promise here, we don't need to wait for this to complete.
And we also don't need the pendingOperationsRef. Just send the message and the transition to connect. No need to return session id. only to send the CreateSession command